### DISCUSSION TITLE
Can I have multiple service workers in my PWA?

### DISCUSSION THREAD

**Discussion by kylereblora** [DuCanhGH/next-pwa#78] [Q&A]
Hi @DuCanhGH! From the docs, next-pwa supports the configuration of custom service workers. Is it possible to have multiple workers in the `customWorkerSrc` path? Or if it isn't, does the custom worker override the default service worker from workbox?

Thanks so much for the library.

**Comment by DuCanhGH** [comment]
You can only have one custom worker, but you can import modules into it. Also, you can use `workboxOptions.swSrc` to opt into `InjectManifest` and create your own service worker :)

**Reply by kylereblora** [reply]
@DuCanhGH Thanks! If I specify a custom worker, does that override the behaviors of the default workbox worker?

**Reply by DuCanhGH** [reply]
@kylereblora that custom worker is bundled and then imported into Workbox's service worker using `importScripts`. I don't know whether you can override its behaviour or not, though.

**Reply by kylereblora** [reply]
Got it, so I think this means that when using a custom worker, this is "appended" to the workbox's service worker.

**Reply by DuCanhGH** [reply]
@kylereblora yeah, that's pretty much it :)
